src/model/price-scale.ts

```diff
@@ -541,6 +541,26 @@ export class PriceScale {
 		return this._mainSource;
 	}
 
+	public firstValue(): number | null {
+		// TODO: cache the result
+		let result: FirstValue | null = null;
+
+		for (const source of this._dataSources) {
+			if (source instanceof PriceDataSource) {
+				const firstValue = source.firstValue();
+				if (firstValue === null) {
+					continue;
+				}
+
+				if (result === null || result.timePoint > firstValue.timePoint) {
```

Comment:
What do you think about swapping the condition to the following?
```suggestion
				if (result === null || firstValue.timePoint < result.timePoint) {
```

Reply:
Looks good